packages/lisk-transactions/src/transactions/base.ts

```diff
@@ -376,33 +394,47 @@ export abstract class BaseTransaction {
 	public apply(sender: Account): TransactionResponse {
 		const updatedBalance = new BigNum(sender.balance).sub(this.fee);
 		const updatedAccount = { ...sender, balance: updatedBalance.toString() };
+		const errors = updatedBalance.gte(0)
+			? []
+			: [
+					new TransactionError(
+						`Account does not have enough LSK: ${sender.address}, balance: ${
+							sender.balance
+						}`,
+						this.id,
+					),
+			  ];
 
 		return {
 			id: this.id,
-			status: Status.OK,
-			state: [updatedAccount],
-			errors: [],
+			status: errors.length > 0 ? Status.FAIL : Status.OK,
+			state: errors.length > 0 ? { sender } : { sender: updatedAccount },
```

Comment:
We should return the updatedAccount whether irrespective of errors. What do you think @shuse2